runtime/databricks/automl_runtime/sklearn/datetime_imputer.py

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright (C) 2022 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pandas as pd
+from sklearn.base import TransformerMixin, BaseEstimator
+
+class DatetimeImputer(TransformerMixin, BaseEstimator):
+    """Imputer for date and timestamp data."""
+
+    def __init__(self, strategy='median', fill_value=None):
```

Comment:
How about we default to mean instead of median? I think they're both reasonable defaults, but it would be easier for us and users to remember if we keep it consistent with the default method used for numerical values. Otherwise this would be the only data type that default to median.

We discussed this briefly during design
https://docs.google.com/spreadsheets/d/1Tz5sIn-FF7dY-comdAtkr4bU15hpxnpbn6EOZgvBP8A/edit#gid=1056641222

Reply:
The reason I think median makes more sense is when the input is date (not timestamp), it is like a discrete, ordered datatype. Using median will still get us a date, but using mean will return a timestamp in between dates. WDYT?

Reply:
Chatted offline and updated the strategy to `mean`.